library/entropy_poll.c

```diff
@@ -44,6 +44,11 @@
  *  **********
  */
 
+#if defined(__linux__)
+/* Ensure that syscall() is available even when compiling with -std=c99 */
+#define _GNU_SOURCE
```

Comment:
As reported in https://github.com/ARMmbed/mbedtls/issues/3432, `_GNU_SOURCE` should not be redefined if it's already defined.

This also applies to `_POSIX_C_SOURCE`, `_XOPEN_SOURCE` and any similar macro.